### PR TITLE
Indexer without elrond go dependency

### PR DESCRIPTION
**Pull request by miiu96**
- latest indexer from `master` branch without elrond-go and arwens dependencies 
- moved TransactionWithFeeHandler in `elrond-go-core repository`

### CODE REVIEW

File: factory/processComponents.go [hunk 1]
```diff
@@ -733,7 +733,12 @@ func (pcf *processComponentsFactory) indexGenesisAccounts() error {
 		genesisAccounts = append(genesisAccounts, userAccount)
 	}
 
-	pcf.statusComponents.ElasticIndexer().SaveAccounts(uint64(pcf.coreData.GenesisNodesSetup().GetStartTime()), genesisAccounts)
+	genesisAccountsData := make([]data.UserAccountHandler, 0, len(genesisAccounts))
+	for idx := 0; idx < len(genesisAccounts); idx++ {
```
Comment (bogdan-rosianu) [inline factory/processComponents.go:737]: no need to iterate 2 times. add this inside the above for

File: epochStart/mock/builtInCostHandlerStub.go [hunk 2]
```diff
@@ -1,20 +1,20 @@
...
+	"github.com/ElrondNetwork/elrond-go-core/data"
 )
 
 // BuiltInCostHandlerStub -
 type BuiltInCostHandlerStub struct {
 }
 
 // ComputeBuiltInCost -
-func (b *BuiltInCostHandlerStub) ComputeBuiltInCost(_ process.TransactionWithFeeHandler) uint64 {
+func (b *BuiltInCostHandlerStub) ComputeBuiltInCost(_ data.TransactionWithFeeHandler) uint64 {
```
Comment (bogdan-rosianu) [inline epochStart/mock/builtInCostHandlerStub.go:12]: might move this interface somewhere else

**Review by bogdan-rosianu — CHANGES_REQUESTED** [review]

File: factory/processComponents.go (hunk 1 above)
Comment (miiu96) [reply factory/processComponents.go]: done.

File: state/baseAccount.go [hunk 3]
```diff
@@ -38,6 +39,14 @@ func (ba *baseAccount) DataTrieTracker() DataTrieTracker {
 	return ba.dataTrieTracker
 }
 
+func (ba *baseAccount) RetrieveValueFromDataTrieTracker(key []byte) ([]byte, error) {
```
Comment (iulianpascalau) [inline state/baseAccount.go:42]: missing comment

Comment (miiu96) [reply state/baseAccount.go:43]: done.

**Review by iulianpascalau — APPROVED** [review]

**Review by bogdan-rosianu — APPROVED** [review]

**Review by bogdan-rosianu — APPROVED** [review]

**Review by bogdan-rosianu — APPROVED** [review]

**Review by gabi-vuls — APPROVED** [review]
System test passed. 
image

**Review by iulianpascalau — APPROVED** [review]

**Pull request merged**